src/main/java/com/eternalcode/parcellockers/gui/SentParcelsGUI.java

```diff
@@ -52,25 +54,27 @@ public void showSentParcelsGUI(Player player) {
             gui.setItem(slot, backgroundItem);
         }
 
-        for (Parcel parcel : this.repository.findBySender(player.getUniqueId()).join()) {
-            ParcelMeta meta = parcel.meta();
-            String sender = player.getName();
-            String receiver = Bukkit.getPlayer(meta.getReceiver()).getName();
+        this.parcelDatabaseService.findBySender(player.getUniqueId()).whenComplete((parcels, throwable) -> {
+            for (Parcel parcel : parcels) {
+                ParcelMeta meta = parcel.meta();
+                String sender = player.getName();
+                String receiver = this.server.getPlayer(meta.getReceiver()).getName();
 
-            gui.addItem(ItemBuilder.from(Material.BUNDLE)
-                    .name(this.miniMessage.deserialize(meta.getName()))
-                    .lore(this.miniMessage.deserialize("&3UUID: ").append(Component.text(parcel.uuid().toString())),
-                            this.miniMessage.deserialize("&3Sender: ").append(Component.text(sender)),
-                            this.miniMessage.deserialize("&3Receiver: ").append(Component.text(receiver)),
-                            this.miniMessage.deserialize("&3Size: ").append(Component.text(meta.getSize().toString())),
-                            this.miniMessage.deserialize("&3Priority: ").append(Component.text(meta.isPriority() ? "&aYes" : "&cNo")),
-                            this.miniMessage.deserialize("&3Description: ").append(Component.text(meta.getDescription())),
-                            this.miniMessage.deserialize("&3Entry position: ").append(Component.text(meta.getEntryLocker().getPosition().toString())),
-                            this.miniMessage.deserialize("&3Target Position: ").append(Component.text(meta.getDestinationLocker().getPosition().toString())),
-                            this.miniMessage.deserialize("&3Recipients: ").append(Component.text(meta.getRecipients().stream().map(Bukkit::getPlayer).map(HumanEntity::getName).toList().toString())))
-                    .flags(ItemFlag.HIDE_ATTRIBUTES)
-                    .asGuiItem());
-        }
-        gui.open(player);
+                gui.addItem(ItemBuilder.from(Material.BUNDLE)
+                        .name(this.miniMessage.deserialize(meta.getName()))
+                        .lore(this.miniMessage.deserialize("&3UUID: ").append(Component.text(parcel.uuid().toString())),
+                                this.miniMessage.deserialize("&3Sender: ").append(Component.text(sender)),
+                                this.miniMessage.deserialize("&3Receiver: ").append(Component.text(receiver)),
```

Comment:
Z doświadczenia wiem że szybciej idzie zrobić to pod config niż się pierdolić z pisaniem tego wszyskiego (lore)